\documentclass[english,12pt]{article}
\usepackage{fontspec}
\setmainfont{FreeSerif}
\setsansfont{FreeSans}
\setmonofont{FreeMono}
\setmathrm{FreeSerif}
\setmathsf{FreeSans}
\setmathtt{FreeSerif}
\usepackage[letterpaper,left=1in,right=1in,top=1in,bottom=1in,headheight=14pt]{geometry}
\usepackage{url}
\usepackage{siunitx}
\usepackage{fancyhdr}
\usepackage{graphicx}
\usepackage[bf]{caption}
\usepackage{rotating}
\usepackage{wrapfig}
\usepackage{booktabs}
\usepackage{float}
\usepackage{multirow}
\usepackage{setspace}
\usepackage{amsmath}
\usepackage[macros=false,single=true,single-form=long]{acro}
\usepackage{array}
\usepackage{dcolumn}
\usepackage[export]{adjustbox}
\usepackage{longtable}
\usepackage{pdflscape}
\usepackage[backend=biber,natbib=true,hyperref=true,citestyle=numeric-comp,style=nature,autocite=inline]{biblatex}
\addbibresource{references.bib}
\usepackage[noblocks,auth-sc]{authblk}
\usepackage[hyperfigures,bookmarks,colorlinks]{hyperref}
%\usepackage[hyperfigures,bookmarks,colorlinks,citecolor=black,filecolor=black,linkcolor=black,urlcolor=black]{hyperref}
% Floats at end for journals
% \usepackage[nolists,figuresfirst,nomarkers,heads,nofighead,notabhead]{endfloat}
% \AtBeginFigures{\section*{Figure Legends}\renewcommand{\efloatseparator}{\hbox{}}}
% \AtBeginTables{\section*{Tables}\renewcommand{\efloatseparator}{\mbox{}}}
\usepackage[nomargin,inline,draft]{fixme}
\newcommand{\DLA}[1]{\textcolor{red}{\fxnote{DLA: #1}}}
% \usepackage[x11names,svgnames]{xcolor}
\usepackage{texshade}
\usepackage{tikz}
\usepackage{nameref}
\usepackage{xspace}
\usepackage[markifdraft,raisemark=0.01\paperheight,draft]{gitinfo2}
% this package will stretch the font slightly if necessary.
%\usepackage{microtype}
\usepackage{zref-xr,zref-user}
\usepackage[capitalize]{cleveref}
\renewcommand{\textfraction}{0.15}
\renewcommand{\topfraction}{0.85}
\renewcommand{\bottomfraction}{0.65}
\renewcommand{\floatpagefraction}{0.60}
\oddsidemargin 0.0in 
\textwidth 6in
\raggedbottom
\clubpenalty = 10000
\widowpenalty = 10000
\setlength{\emergencystretch}{20em}
% make the afill smaller
\renewcommand\Affilfont{\itshape\small}
\renewcommand\cite[1]{\autocite{#1}}
\pagestyle{fancy}
\ExplSyntaxOn
\NewAcroCommand \gene {
  \group_begin:
    \keys_set:nn {acro} {short-format = \itshape}
    \acro_use:n{#1}
  \group_end:
}
\NewAcroCommand \protein { \acro_use:n {#1} }
\ExplSyntaxOff

\input{acronyms}

\begin{document}
\author[1]{Don Armstrong}
\author[1,2]{Monica Uddin}
\author[1,3,*]{Derek E. Wildman}
\title{Human tissue specific markers}
\affil[1]{Institute for Genomic Biology, University of Illinois at Urbana-Champaign, Urbana, IL, USA}
\affil[2]{Department of Psychology, University of Illinois at Urbana-Champaign, Urbana, IL, USA}
\affil[3]{Department of Molecular and Integrative Physiology, University of Illinois at Urbana-Champaign, Urbana, IL, USA}
\affil[*]{To whom correspondence should be addressed}
\date{\today}

\maketitle

\doublespacing

<<load.libraries,echo=FALSE,results="hide",warning=FALSE,message=FALSE,error=FALSE,cache=FALSE>>=
opts_chunk$set(out.width="0.5\\textwidth",out.height="0.7\\textheight",out.extra="keepaspectratio")
opts_chunk$set(dev="cairo_pdf",cache=TRUE, autodep=TRUE)
options(scipen = -1, digits = 2)
library("data.table")
library("ggplot2")
library("reshape2")
library("proto")
library("xtable")
library("caret")
library("pROC")
library("grid")
library("scales")

capwords <- function(s, strict = FALSE) {
    cap <- function(s) paste(toupper(substring(s, 1, 1)),
                             {s <- substring(s, 2); if(strict) tolower(s) else s},
                             sep = "", collapse = " " )
    sapply(strsplit(s, split = " "), cap, USE.NAMES = !is.null(names(s)))
}

@ 


<<load.data,echo=FALSE,results="hide",warning=FALSE,cache.extra=file.info(paste0("../tissue_specific_expression/",c("tissue_specific_markers","interesting_gene_reads_wide","combined_star_logs","svm_test_results","svm_test_results_reduced","housekeeping_genes_superset")))[,"mtime"]>>=
for (a in paste0("../tissue_specific_expression/",
                 c("tissue_specific_markers",
                   "interesting_gene_reads_wide",
                   "combined_star_logs",
                   "svm_test_results",
                   "svm_test_results_reduced",
                   "housekeeping_genes_superset"
                   ))) {
    load(a)
}
tissue.specific.genes <- data.table(tissue.specific.genes)
setkey(tissue.specific.genes,"gene")

tissue.specific.genes[,type:="not specific"]
tissue.specific.genes[,housekeeping:="not housekeeping"]
tissue.specific.genes[possibly.specific==TRUE,type:="tissue specific"]
tissue.specific.genes[tracking_id %in% housekeeping.genes.superset[,tracking_id] &
                      possibly.specific==FALSE,housekeeping:="housekeeping"]
tissue.specific.genes[tracking_id %in% housekeeping.genes.superset[,tracking_id] &
                      possibly.specific==TRUE,housekeeping:="housekeeping and specific"]

tissue.specific.genes[,type:=
                          factor(type,
                                 levels=c("not specific","tissue specific",
                                     "housekeeping",
                                     "housekeeping and specific"
                                          ))]
@ 

<<genes.table.sanitize,echo=FALSE,results="hide">>=
genes.table.sanitize <- function(str){
    return(str)
}
link.gene.names <- function(gene){
    return(paste0("\\href{http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=",
                  gene,"}{",
                  "\\textit{",xtable.sanitize.text(gene),
                  "}}"))
}
xtable.sanitize.text <- function(str){
    result <- str
    result <- gsub("\\\\", "SANITIZE.BACKSLASH", result)
    result <- gsub("$", "\\$", result, fixed = TRUE)
    result <- gsub(">", "$>$", result, fixed = TRUE)
    result <- gsub("<", "$<$", result, fixed = TRUE)
    result <- gsub("|", "$|$", result, fixed = TRUE)
    result <- gsub("{", "\\{", result, fixed = TRUE)
    result <- gsub("}", "\\}", result, fixed = TRUE)
    result <- gsub("%", "\\%", result, fixed = TRUE)
    result <- gsub("&", "\\&", result, fixed = TRUE)
    result <- gsub("_", "\\_", result, fixed = TRUE)
    result <- gsub("#", "\\#", result, fixed = TRUE)
    result <- gsub("^", "\\verb|^|", result, fixed = TRUE)
    result <- gsub("~", "\\~{}", result, fixed = TRUE)
    result <- gsub("SANITIZE.BACKSLASH", "$\\backslash$",
                   result, fixed = TRUE)
    return(result)
}
source("/home/don/projects/xtable/pkg/R/print.xtable.R")

@ 

<<compact_tissue_name,echo=FALSE,results="hide">>=
compact_tissue_name <- function(tissue) {
    gsub("nhlf","human lung fibroblasts",
         gsub("cultured cells","",
         tissue))
}
@ 

<<specific.markers.plot.function,echo=FALSE,results="hide",warning=FALSE>>=
specific.markers.plot <- function(tissue.name,genes) {
    setkey(interesting.gene.reads.wide,"gene_short_name")
    if (missing(genes))
        genes <- as.character(tissue.specific.genes[tissue==tissue.name,gene])
    interesting.tissues <- 
        colnames(interesting.gene.reads.wide)[
         apply(interesting.gene.reads.wide[genes,-(1:2),with=FALSE],1,
                             function(x){order(-x)[1:2]+2})]
    interesting.tissues <-
        c(tissue.name,
          interesting.tissues)
    interesting.tissues <- 
        c("gene_short_name",unique(interesting.tissues))
    
    tissue.specific.gene.reads <- 
        data.table(melt(interesting.gene.reads.wide[genes,interesting.tissues,with=FALSE],
                        id="gene_short_name",
                        variable.name="tissue",
                        value.name="FPKM"
                        ))
    tissue.specific.gene.reads[,compact_tissue:=factor(compact_tissue_name(tissue),
                                                       levels=compact_tissue_name(interesting.tissues))
                               ]
    tissue.specific.gene.reads[,`log_2(FPKM)`:=log2(FPKM+1)]
    if (!is.null(tissue.name)) {
        gene.levels <- 
            unique(as.character(tissue.specific.genes[tissue==tissue.name][order(specificity),gene]))
    } else {
        gene.levels <- 
            unique(as.character(tissue.specific.genes[tissue %in% interesting.tissues
                                                      ][order(specificity),gene]))
    }
    tissue.specific.gene.reads[,gene_short_name:=factor(gene_short_name,levels=gene.levels)]
    return(ggplot(tissue.specific.gene.reads,
                  aes(y=gene_short_name,
                      x=compact_tissue))
           + geom_tile(aes(fill=`log_2(FPKM)`))
           + scale_fill_gradient(limits=c(0,17),
                                 low="white",
                                 high="black",
                                 guide=guide_colourbar(title=expression(log[2](FPKM))))
           + scale_x_discrete("", expand = c(0, 0))
           + scale_y_discrete("", expand = c(0, 0))
           + theme_grey(base_size=14)
           + theme(legend.position = "top",
                   axis.ticks = element_blank(),
                   panel.grid=element_blank(),
                   panel.border=element_blank(),
                   panel.background=element_blank(),
                   plot.background=element_blank(),
                   plot.margin=unit(c(1,1,1,1),"mm"),
                   axis.text.y = element_text(size=10,face="italic"),
                   axis.text.x = element_text(size=9,angle = 30, hjust = 1, vjust=1))
           )
}
@ 

<<specific_markers_barplot,echo=FALSE,results="hide",warning=FALSE>>=
specific.markers.barplot <- function(tissue.name,genes) {
    setkey(interesting.gene.reads.wide,"gene_short_name")
    if (missing(genes))
        genes <- as.character(tissue.specific.genes[tissue==tissue.name,gene])
    tissue.specific.gene.reads <- 
        data.table(melt(interesting.gene.reads.wide[genes,],
                        id=c("gene_short_name","tracking_id"),
                        variable.name="tissue",
                        value.name="FPKM"
                        ))
    tissue.specific.gene.reads[,`log_2(FPKM)`:=log2(FPKM+1)]
    return(ggplot(tissue.specific.gene.reads,
                  aes(x=gene_short_name,
                      y=FPKM+1))
           + geom_violin()
           + scale_x_discrete("Gene")
           + scale_color_discrete("Tissue")
           + scale_shape_discrete("Tissue")
           + scale_y_log10()
           + geom_jitter(data=tissue.specific.gene.reads[order(-FPKM),
                                                        list(tissue=tissue[1:3],
                                                             max_fpkm=pmax(FPKM)[1:3]),
                                                        by=gene_short_name],
                        aes(y=max_fpkm+1,
                            x=gene_short_name,
                            color=capwords(as.character(tissue)),
                            shape=capwords(as.character(tissue))
                            ),
                        size=2
                        )
           + coord_flip()
           )
}
@ 


\section{Abstract}

In this paper we want to show that given a population of \ac{EV}

\textbf{Can we determine the proportions and identities of
  the tissue(s) of \ac{EV} origin?}

\textbf{Can assign differential expression to a particular tissue of
  origin in \ac{EV}?}

\textbf{Housekeeping genes are not tissue specific}

\textbf{Additional genes which are likely to be housekeeping but were
  not identified by previous papers}

\section{Introduction}

\subsection{\ac{EV} background}

\paragraph{What are \acsp{EV}}

\Ac{EV} are small (40-1000nm) compartments that can be found in
blood, saliva, and urine. \Ac{EV} originate from the plasma membrane
or multivesicular endosome of many cell types and contain proteins,
lipids, mRNA, miRNA, and many other non-coding RNAs. Because \ac{EV}
contain mRNA from the cell of origin and can be obtained without
invasive procedures, they are a promising remote sensor of the
transcriptome of tissues that would otherwise be inaccessible.

\paragraph{Where are \acsp{EV} found?}

\paragraph{How are \acsp{EV} formed and what are their contents}


\subsection{\acsp{EV} will enable routine liquid biopsies}

\paragraph{Existing work showing promise of \acsp{EV}}

\paragraph{Existing methods require tumor/disease specific marker which is unique to that tissue}

\subsection{Identifying tissues of origin}

\paragraph{Existing protein surface markers}

\DLA{Is there a database of tissue/cell specific markers?}

\paragraph{Issues of single cell transcriptome heterogeneity}

\paragraph{Tissue specificity index}


\subsection{Assigning differential expression}

\subsection{Housekeeping genes}

\subsection{Cell free RNA and Exosomes in Diagnostics}

One requirement for remotely sensing the transcriptome is identifying
the source tissue of a specific \ac{EV}. To identify the source tissue
of a specific \ac{EV}, we have identified miRNA and mRNA markers which
are characteristic of multiple organs and tissues from existing
publicly-available RNAseq transcriptomes by calculating a tissue
specificity index.

    
\section{Methods}

\subsection{Tissue RNA-seq}

\subsubsection{Samples}
\Sexpr{star.logs[grepl("Number of input reads",field),sum(as.numeric(value))]} %
%
RNAseq reads from %
%
\Sexpr{length(star.logs[grepl("Number of input reads",field),value])} %
% 
samples (supplemented with FPKMs from
\Sexpr{nrow(predictions.caret.svm[["uterus"]][["training"]])+
  nrow(predictions.caret.svm[["uterus"]][["testing"]]) -
  length(star.logs[grepl("Number of input reads",field),value])} %
%
additional samples without raw data in the \ac{SRA}) corresponding
to %
\Sexpr{length(unique(na.omit(tissue.specific.genes[,tissue])))} %
different tissues were obtained from the Roadmap Epigenomics
\cite{Bernstein.ea2010:NIHRoadmapEpigenomicsMapping} and GTEx
\cite{Gtexconsortium.2015:Thegenotypetissueexpression} projects, as
well as uterine (GSE50599,
\cite{Chan.ea2014:Assessmentofmyometrialtranscriptome}) and placenta
(GSE66622, \cite{Hughes.ea2015:Evaluatingintra-andinter-individual})
samples from the \ac{SRA}.

\subsubsection{Alignment and Calling}
Reads were aligned to \textit{Homo sapiens} reference genome GRCh38
with Ensembl annotation release 80 using STAR v2.4.2a
\cite{Dobin.ea2013:STARultrafastuniversalRNA-seq} and called using
Cufflinks v2.2.1
\cite{Trapnell.ea2009:TopHatdiscoveringsplicejunctions}. The average
percentage of uniquely mapped reads was %
\Sexpr{mean(as.numeric(gsub("%","",star.logs[field=="Uniquely mapped reads %",value])))}\%
and the average total mapping percentage was %
\Sexpr{mean(as.numeric(gsub("%","",star.logs[field=="Uniquely mapped reads %",value]))+as.numeric(gsub("%","",star.logs[field=="% of reads mapped to multiple loci",value])))}\%.
% 
For samples where the reads were not publicly available (GTEx and
some Roadmap Epigenomics samples), the RPKM were taken from publicly
available result files (see code for details). 

\subsubsection{Normalization}

\subsection{Tissue Specific Markers}

\subsubsection{Tissue Specificity Index}

The tissue specificity index for gene $g$ ($\tau_g$) is
    
\begin{equation}
  \tau_g = \frac{\sum_{i=1}^{i=N} \left(1-\frac{X_{i,g}}{\mathrm{max}(X_g)}\right)}{N-1}
\end{equation}
\label{eq:tissue_specificity_index}
    
where $N$ is the number of tissues, $X_{i,g}$ is the mean expression
of gene $g$ in tissue $i$ and $\mathrm{max}(X_g)$ is the maximal
expression of gene $g$ across all tissues.
    
We also trained a multiclass \ac{SVM} to accurately identify the
tissue of origin, and present the results of the identification of
reads from uterine and placental origin here. Tables showing tissue
specific genes from all
\Sexpr{length(unique(na.omit(tissue.specific.genes[,tissue])))}
tissues can be found on \url{http://dla2.us/p/em2015}.

\subsection{Read Biasing}

The population of RNAs found in \ac{EV} are distinct from the cytosolic
pool, and are likely selectively sorted into \ac{EV}~%
\cite{Villarroya-Beltri.ea2014:Sortingitoutregulation,Hoen.ea2012:DeepsequencingofRNA,Janas.ea2015:MechanismsofRNAloading,Balkom.ea2015:Quantitativeandqualitativeanalysis}.
Specifically, the motives
GGAG~\cite{Villarroya-Beltri.ea2013:SumoylatedhnRNPA2B1controlssorting},
ACCAGCCU, CAGUGAGC, and
UAAUCCCA~\cite{Batagov.ea2013:Exosomessecretedbyhuman} are enriched in
\ac{EV}. Owing to the limited number of tissue-specific \ac{EV} RNA-seq
results, we have biased the reads from existing tissue-specific
RNA-seq results to preferentially include RNAs which contain motifs
found in exosomes and also match the lower number of unique RNAs which
are found in individual exosomes.

\subsection{Machine Learning}

The classification of samples into reads of placental origin vs
non-placental origin or uterine origin vs non-uterine origin was
performed using a \ac{SVM}. The \ac{SVM} was generated using caret
\cite{Kuhn2008:Buildingpredictivemodelsinr} and svmRadial
\cite{Karatzoglou2004:kernlab} with 10 cycles of 10 repeats and a tune
length of 8 with a training set representing $\approx 75\%$ of the
samples ($\approx 2450$ samples total, including $\approx 35$ uterine
or $\approx 60$ placenta) in each of the groups chosen at random, with
the remaining samples making up a testing set ($\approx 820$ samples).


\section{Results}

\subsection{Tissue Specific Genes and Housekeeping genes}

\begin{figure}
<<tissue.specificity.shape.figure,fig.width=4,fig.height=4,echo=FALSE>>=
print(ggplot(tissue.specific.genes[gi.max>=1],
             aes(y=specificity,x=gi.max))+
      ##geom_point()+
      geom_hex(bins=50)+
      geom_line(aes(x=x,y=y),inherit.aes=FALSE,
                color="orange",
                data=data.frame(x=c(10,10,tissue.specific.genes[,max(gi.max)]),
                                y=c(1,0.98,0.98)))+
      ## stat_summary_2d(fun=function(x){length(x)},bins=100)+
      geom_point(aes(colour=gsub("And","\nAnd",
                                 capwords(housekeeping))),
                 size=2,
                 data=tissue.specific.genes[housekeeping!="not housekeeping",])+
      scale_x_log10(breaks = trans_breaks("log10", function(x) 10^x),
                    labels = trans_format("log10", math_format(10^.x)))+
      scale_color_brewer("Housekeeping",palette="Set2")+
      scale_fill_continuous("Count")+
      xlab("Max Expression (FPKM)")+
      ylab("Tissue Specificity Index")+
      theme(legend.position=c(1,0),
            legend.justification=c(1,0),
            legend.box.just="right"
            )
    )

@ 
\caption{Tissue specificity index versus maximum expression in one
  tissue. Genes above and to the right of the orange lines are tissue
  specific. Genes which have been previously
  identified~\cite{Eisenberg.ea2013:Humanhousekeepinggenesrevisited}
  as housekeeping genes are generally not tissue specific, with the
  exception of %
  %
  \Sexpr{tissue.specific.genes[housekeeping=="housekeeping and specific",gene]}.%
  %
}
\label{fig:tissue_specific_shape}
\end{figure}

\Cref{fig:tissue_specific_shape} shows the shape of the tissue
specificity measure of all genes along with the maximum expression in
a particular tissue. Almost all housekeeping genes identified by
\citet{Eisenberg.ea2013:Humanhousekeepinggenesrevisited} have
$\tau_g \le 0.93$ with the exception of
%
\Sexpr{tissue.specific.genes[housekeeping=="housekeeping and specific",gene]}
%

\subsubsection{Genes which are housekeeping and specific}

\DLA{Why are these two genes housekeeping and specific?}

\begin{figure}
<<housekeeping_tissue_specific_markers_figure,fig.width=4,fig.height=4,echo=FALSE>>=
print(specific.markers.barplot(genes=tissue.specific.genes[housekeeping=="housekeeping and specific",gene])
      + xlab("Housekeeping Tissue\nSpecific Genes")
      + theme(legend.position="Top")
      + ylab("Tissue"))
@
\caption{Genes which are tissue specific but are also housekeeping genes}
\label{fig:housekeeping_tissue_specific_genes}
\end{figure}

<<tissue.specific.markers.table,results="hide",echo=FALSE>>=
tissue.specific.markers.table <- function(tissue,legend,min.gi=0) {
    this.tissue <- tissue
    temp <-
        tissue.specific.genes[tissue==this.tissue & gi.max >= min.gi,
                              list(Gene=link.gene.names(gene),
                                   "{Tissue specificity}"=specificity,
                                   "{Mean Expression}"=gi.max)]
    setorder(temp,-`{Mean Expression}`)
    print(xtable(temp,
                 align=c("r","c","S","S[table-format=1.0]"),
                 caption=legend,
                 label=paste0("tab:",tissue,"_specific_genes")),
          sanitize.text.function=genes.table.sanitize,
          include.rownames=FALSE,
          booktabs=TRUE,
          floating=TRUE,
          )
    rm(temp)
}

@ 


\subsection{Uterus and Placenta Specific Genes}

<<uterus.specific.markers.table,results="asis",echo=FALSE>>=
tissue.specific.markers.table(tissue="uterus",
                              legend="Uterus specific genes")
@ 

\begin{figure}
<<uterus.specific.markers.figure,fig.width=4,fig.height=4,echo=FALSE>>=
print(specific.markers.plot("uterus")
      + xlab("Genes Characteristic of Uterus ")
      + ylab("Tissue"))
@
\caption{Expression of genes characteristic of uterus
  ($\tau_g \ge 0.98$) in uterus and all tissues with the second
  highest expression for any of the uterus-characterizing genes
  ordered by their tissue specificity. The known uterus-characterizing
  gene PAEP and short RNAs ZNFX1-AS1\_3 (87bp) and AL356310.1 (94bp)
  are highly characteristic of uterus.}
\label{fig:uterus_specific_markers}
\end{figure}

FTX\_1 is very interseting

PCP4 is troubling, because this is a purkingie cell protein

Some of the markers which we had previously seen (and in fact mention
in our abstract have gone away).

We originally identified MIR1914, MIR126, SNORD12, and SNORD5 as
uterus specific. MIR1914 is actually placenta specific, as is MIR126.

This occured because the placenta FPKMs we based the original list on
did not properly align against miRNA or snRNA
\cref{tab:missing_uterus_specific_genes}. The uterus calls also appear
to be wrong or cooked; for example, SNORD5 has counts of 7000ish which
are not inside of its confidence interval. This may be indicative of a
off-by-one error or something else that is screwed up in their file.

<<missing.uterus.specific.markers.table,results="asis",echo=FALSE>>=
temp <- tissue.specific.genes[c("MIR1914","MIR126","SNORD12","SNORD5"),
                              list(gene,tissue,specificity,gi.max)]
setnames(temp,c("Gene","Tissue","Tissue Specificity","Mean Expression"))
print(xtable(temp,
             caption="Missing previously uterus specific genes",
             label="tab:missing_uterus_specific_genes"),
      include.rownames=FALSE,
      )
rm(temp)
@ 

<<placenta.specific.markers.table,results="asis",echo=FALSE>>=
tissue.specific.markers.table(tissue="placenta",
                              legend=
                                  paste0("Placenta specific genes with expression ",
                                         "$\\ge 100$ ordered by median expression in placenta."),
                              min.gi=100
                              )
@ 

\begin{figure}
<<placenta.specific.markers.figure,fig.width=4,fig.height=6,echo=FALSE>>=
print(specific.markers.plot("placenta",
                            genes=as.character(tissue.specific.genes[tissue=="placenta" & 
                                                                         gi.max >= 1024
                               ,gene]))
      + xlab("Genes Characteristic of Placenta")
      + ylab("Tissue"))
@

}
\caption{Expression of genes characteristic of placenta
  ($\tau_g \ge 0.98$) with high expression ($\mathrm{FPKM} \ge 1024$)
  in placenta and all tissues with the second highest expression for
  any placenta-characterizing gene ordered by their specificity to
  placenta. The previously known placenta-characterizing PSG genes and
  CSH genes are nearly unique to placenta and the short RNAs mir-126,
  mir-622, and PAPPA-AS1 are less so.}
\label{fig:placenta_specific_markers}
\end{figure}

\subsection{Biasing reads towards exosomes}

\subsection{Machine learning to identify tissue source}

\begin{figure}
<<placenta.roc.curve,fig.width=8,fig.height=8,echo=FALSE>>=
layout(matrix(c(1,3,2,4),2,2))
pl.roc.obj1 <- 
    plot.roc(as.numeric(predictions.caret.svm[["placenta"]][["testing"]][,"actual"])-1,
             as.numeric(predictions.caret.svm[["placenta"]][["testing"]][,"placenta"]),
             percent=TRUE,
             ## main="Training vs Testing set for Uterus",
             col="#002058"
             )
pl.roc.obj2 <- 
    lines.roc(as.numeric(predictions.caret.svm[["placenta"]][["training"]][,"actual"])-1,
              as.numeric(predictions.caret.svm[["placenta"]][["training"]][,"placenta"]),
              percent=TRUE,
              col="#E87722")
# testobj <- roc.test(rocobj1, rocobj2)
# text(50, 50, labels=paste("p-value =", format.pval(testobj$p.value)), adj=c(0, .5))

legend("bottomright",
       legend=c("Placenta Testing Set", "Placenta Training Set"),
       col=c("#002058", "#E87722"),
       lwd=2)

ut.roc.obj1 <- 
    plot.roc(as.numeric(predictions.caret.svm[["uterus"]][["testing"]][,"actual"])-1,
             as.numeric(predictions.caret.svm[["uterus"]][["testing"]][,"uterus"]),
             percent=TRUE,
             ## main="Training vs Testing set for Uterus",
             col="#002058"
             )
ut.roc.obj2 <- 
    lines.roc(as.numeric(predictions.caret.svm[["uterus"]][["training"]][,"actual"])-1,
              as.numeric(predictions.caret.svm[["uterus"]][["training"]][,"uterus"]),
              percent=TRUE,
              col="#E87722")
# testobj <- roc.test(rocobj1, rocobj2)
# text(50, 50, labels=paste("p-value =", format.pval(testobj$p.value)), adj=c(0, .5))

legend("bottomright",
       legend=c("Uterus Testing Set", "Uterus Training Set"),
       col=c("#002058", "#E87722"),
       lwd=2)

for (a in c("placenta","uterus")) {
pl.roc.obj1 <- 
    plot.roc(as.numeric(predictions.caret.svm.reduced[[a]][["testing.1000.reads"]][,"actual"])-1,
             as.numeric(predictions.caret.svm.reduced[[a]][["testing.1000.reads"]][,a]),
             percent=TRUE,
             ## main="Training vs Testing set for Uterus",
             col="#002058"
             )
pl.roc.obj2 <- 
    lines.roc(as.numeric(predictions.caret.svm.reduced[[a]][["testing.5000.reads"]][,"actual"])-1,
              as.numeric(predictions.caret.svm.reduced[[a]][["testing.5000.reads"]][,a]),
              percent=TRUE,
              col="#E87722")
pl.roc.obj3 <- 
    lines.roc(as.numeric(predictions.caret.svm.reduced[[a]][["testing.10.reads"]][,"actual"])-1,
              as.numeric(predictions.caret.svm.reduced[[a]][["testing.10.reads"]][,a]),
              percent=TRUE,
              col="green")
pl.roc.obj4 <- 
    lines.roc(as.numeric(predictions.caret.svm.reduced[["placenta"]][["testing.1.reads"]][,"actual"])-1,
              as.numeric(predictions.caret.svm.reduced[["placenta"]][["testing.1.reads"]][,"placenta"]),
              percent=TRUE,
              col="brown")
# testobj <- roc.test(rocobj1, rocobj2)
# text(50, 50, labels=paste("p-value =", format.pval(testobj$p.value)), adj=c(0, .5))

legend("bottomright",       legend=paste0(capwords(a)," ",c("1 Read","10 Reads","1000 Reads", "5000 Reads")),
       col=c("brown","green","#002058", "#E87722"),
       lwd=2)
}
grid.text(c("A","B","C","D"),
          x=unit(c(0,0.5,0,0.5),"npc"),
          y=unit(c(1,1,0.5,0.5),"npc"),
          just=c("left","top"),
          gp=gpar(fontsize=32)
          )

@

\caption{\ac{ROC} curve of the placenta \ac{SVM} (A,C) and
  uterus \ac{SVM} (B,D) . Both \ac{SVM} are highly effective at
  distinguishing between samples of uterus origin and non-uterus
  origin using the entirety of reads from a sample.
  %
  \ac{ROC} curve of reduced reads in Placenta (C) and
  Uterus (D) samples with the indicated number of total reads. Reads were
  reduced by sampling reads for each sample weighted by the FPKM
  measured for each sample, and run using the same trained SVM
  from the full runs.
  % 
  The placenta \ac{SVM} is fairly accurate at distinguishing between
  samples of placental origin and non-placental origin even when the
  number of reads is very low (10). The uterus \ac{SVM} requires many
  more reads in order to achieve similar accuracy. This is likely due
  to the much smaller number of uterus specific genes in comparison to
  placenta.}
\label{fig:plac_ut_roc}
\end{figure}


<<calc.reduced,echo=FALSE,results="asis">>=
reduced.reads <- function(fpkms,num.reads) {
    non.zero <- fpkms > 0
    ret.val <- rep.int(0,length(fpkms))
    if (!any(non.zero)) {
        return(fpkms)
    }
    inc.prob <- sampling::inclusionprobabilities(fpkms[non.zero],1)
    ret.val[non.zero] <-
        apply(sapply(1:(num.reads/1),function(y){sampling::UPrandomsystematic(inc.prob,0)}),1,sum)
    names(ret.val) <- names(fpkms)
    ret.val
}
set.seed(1021)
reads <- c("A"=1000,
           "B"=2000,
           "C"=20,
           "D"=5000)
reads.to.table <- function(read,reads){
    data.table("Total Reads"=read,
               "Gene A"=reads["A"],
               "Gene B"=reads["B"],
               "Gene C"=reads["C"],
               "Gene D"=reads["D"])
}
rr.table <- list(reads.to.table("Orig. FPKM",reads))
i <- 0
for (read in c(500,500,100,100,5,5,5)){
    i <- i+1
    reads <- reduced.reads(reads,read)
    rr.table[[paste0("read.",read,'.',i)]] <- 
        reads.to.table(read,reads)
}
print(xtable(rbindlist(rr.table),
             digits=0
             ),
      include.rownames=FALSE)
@ 

\section{Discussion}


\section{Conclusion}

  \begin{itemize}
  \item Pregnancy-specific glycoproteins (PSG1, 2, 3 et al.) are
    characteristic of placental tissues, and PAEP is characteristic of
    uterus, as previously described.
%   \item Numerous Long-noncoding RNA and smaller RNAs are specific to uterus and placenta
  \item \ac{SVM} are able to accurately identify the tissue of origin
    for uterus and placenta.
  \item Very few sequenced reads are required to accurately identify
    tissues of Placental origin. We are able to detect 100\% of the
    placenta tissues with 80\% specificity with just 10 reads.
  \item We can easily identify exosomes originating from the placenta,
    as even the smallest exosomes can contain 70 RNA molecules
    \cite{Li.ea2014:AnalysisofRNAcontent}.
  \item Non-maternal variants may also be able to distinguish between
    placental reads and maternal reads.
  \end{itemize}


\printbibliography

\end{document}
