\documentclass[a0paper,landscape]{tikzposter}
\usepackage{fontspec}
\setmainfont{FreeSerif}
\setsansfont{FreeSans}
\setmonofont{FreeMono}
\usepackage{url}
\usepackage{fancyhdr}
\usepackage{graphicx}
\usepackage[bf]{caption}
\usepackage{rotating}
\usepackage{wrapfig}
\usepackage{fancybox}
\usepackage{booktabs}
% \usepackage{multirow}
\usepackage{acronym}
\usepackage[backend=biber,natbib=true,hyperref=true,style=nature]{biblatex}
\addbibresource{references.bib}
% \usepackage[nomargin,inline,draft]{fixme}
% \newcommand{\DLA}[1]{\textcolor{red}{\fxnote{DLA: #1}}}
\usepackage[hyperfigures,bookmarks,colorlinks,citecolor=black,filecolor=black,linkcolor=black,urlcolor=black]{hyperref}
\usepackage{texshade}
\usepackage{tikz}
\usepackage{nameref}
\usepackage{zref-xr,zref-user}

% The textpos package is necessary to position textblocks at arbitary 
% places on the page.  Use showboxes option to show outlines of textboxes.
% \usepackage[absolute]{textpos}
\usepackage[absolute,showboxes]{textpos}

\usepackage{multirow}
\usepackage{array}

\title{Identifying the Tissue of Origin of Extracellular Vesicles Using RNA Expression Signatures}
\author{Don L. Armstrong, Monica Uddin, Derek Wildman}
\institute{Institute for Genomic Biology, Computing Genomes for Reproductive Health, University of Illinois, Urbana-Champaign}
\usetheme{Default}

% use this
% http://www.identitystandards.illinois.edu/graphicstandardsmanual/othermedia/posterpresentation.html
\definecolorstyle{UofI}{
  \definecolor{boldBlue}{HTML}{002058}
  \definecolor{boldOrange}{HTML}{E87722}
  \definecolor{professionalBlue}{HTML}{606EB2}
  \definecolor{professionalOrange}{HTML}{D45D00}
  }{
    \colorlet{backgroundcolor}{professionalBlue}
    \colorlet{framecolor}{black}
    \colorlet{titlefgcolor}{boldBlue}
    \colorlet{titlebgcolor}{white}
    \colorlet{blockbodybgcolor}{white}
    \colorlet{blocktitlebgcolor}{professionalOrange}
    \colorlet{blocktitlefgcolor}{boldBlue}
    }
\usecolorstyle{UofI}
\defineblockstyle{UofIMinimal}{
    titlewidthscale=1, bodywidthscale=1, titleleft,
    titleoffsetx=0pt, titleoffsety=0pt, bodyoffsetx=0pt, bodyoffsety=0pt,
    bodyverticalshift=0pt, roundedcorners=0, linewidth=0.2cm,
    titleinnersep=1cm, bodyinnersep=1cm
}{
    \begin{scope}[line width=\blocklinewidth, rounded corners=\blockroundedcorners]
       \ifBlockHasTitle %
           \draw[draw=none,fill=blockbodybgcolor]
               (blockbody.south west) rectangle (blocktitle.north east);
           \draw[color=blocktitlebgcolor, loosely dashed]
               (blocktitle.south west) -- (blocktitle.south east);%
       \else
             \draw[draw=none,fill=blockbodybgcolor]
                 (blockbody.south west) rectangle (blockbody.north east);
        \fi
    \end{scope}
}

\useblockstyle{UofIMinimal}
\definebackgroundstyle{UofIVertical}{
    \draw[line width=0pt, bottom color=backgroundcolor, 
    top color=white] (bottomleft) rectangle (topright);
}
\usebackgroundstyle{UofIVertical}
\definetitlestyle{UofITitle}{
    width=\paperwidth, roundedcorners=0, linewidth=0pt, innersep=12mm,
    titletotopverticalspace=5mm, titletoblockverticalspace=10mm,
    titlegraphictotitledistance=10pt
}{}
\usetitlestyle{UofITitle}

\makeatletter
\renewcommand\TP@maketitle{%
   \begin{minipage}{0.8\textwidth}
        \color{titlefgcolor}
        {\bfseries \Huge \sc \@title \par}
        \vspace*{1em}
        {\huge \@author \par}
        \vspace*{1em}
        {\LARGE \@institute}
    \end{minipage}%
    \hfill
    \begin{minipage}{0.2\linewidth}
      \includegraphics[width=0.8\linewidth,height=6cm,keepaspectratio]{figures/igb_wordmark}
    \end{minipage}
}
\makeatother

\begin{document}
\acrodef{EV}[EV]{Extracellular Vesicle}
\acrodef{SVM}{Support Vector Machine}
\acrodef{ROC}{Receiver Operating Characteristic}
\acrodef{KNN}{K Nearest Neighbors}

<<load.libraries,echo=FALSE,results="hide",warning=FALSE,message=FALSE,error=FALSE,cache=FALSE>>=
opts_chunk$set(dev="cairo_pdf",out.width="0.20\\columnwidth",out.height="0.7\\textheight",out.extra="keepaspectratio")
#opts_chunk$set(cache=TRUE, autodep=TRUE)
options(device = function(file, width = 8, height = 7, ...) {
  cairo_pdf(tempfile(), width = width, height = height, ...)
})
options(digits=2)
library("data.table")
library("ggplot2")
library("reshape2")
library("proto")
library("grid")
library("xtable")
library("caret")
library("pROC")
@ 

<<load.data,echo=FALSE,results="hide",warning=FALSE,cache.extra=tools::md5sum(paste0("../tissue_specific_expression/",c("tissue_specific_markers","interesting_gene_reads_wide","combined_star_logs","svm_test_results")))>>=
for (a in paste0("../tissue_specific_expression/",
                 c("tissue_specific_markers",
                   "interesting_gene_reads_wide",
                   "combined_star_logs",
                   "svm_test_results"
                   ))) {
    load(a)
}
tissue.specific.genes <- data.table(tissue.specific.genes)
@ 

\maketitle

<<specific.markers.plot.function,echo=FALSE,results="hide",warning=FALSE>>=
specific.markers.plot <- function(tissue.name,genes) {
    setkey(interesting.gene.reads.wide,"gene_short_name")
    if (missing(genes))
        genes <- as.character(tissue.specific.genes[tissue==tissue.name,gene])
    interesting.tissues <- 
        colnames(interesting.gene.reads.wide)[
           which(apply(apply(interesting.gene.reads.wide[genes,-(1:2),with=FALSE],1,
                             function(x){order(-x)==2}),1,any))+2]
     interesting.tissues <- 
        c("gene_short_name",tissue.name,interesting.tissues)
    
    tissue.specific.gene.reads <- 
        data.table(melt(interesting.gene.reads.wide[genes,interesting.tissues,with=FALSE],
                        id="gene_short_name",
                        variable.name="tissue",
                        value.name="FPKM"
                        ))
    tissue.specific.gene.reads[,`log_2(FPKM)`:=log2(FPKM+1)]
    return(ggplot(tissue.specific.gene.reads,
                  aes(y=gene_short_name,x=tissue))
           + geom_tile(aes(fill=`log_2(FPKM)`))
           + scale_fill_gradient(limits=c(0,17),
                                 low="white",
                                 high="#002058",
                                 guide=guide_colourbar(title=expression(log[2](FPKM))))
           + scale_x_discrete("", expand = c(0, 0))
           + scale_y_discrete("", expand = c(0, 0))
           + theme_grey(base_size=14)
           + theme(legend.position = "top",
               axis.ticks = element_blank(), 
               plot.margin=unit(c(1,1,1,1),"mm"),
               axis.text.x = element_text(angle = 30, hjust = 1, vjust=1))
           )
}
@ 


\begin{columns}

  \column{0.25}

  \block{Introduction}{\acp{EV} are small (40-1000nm) compartments
    that can be found in blood, saliva, and urine. \acp{EV} originate
    from the plasma membrane or multivesicular endosome of many cell
    types and contain proteins, lipids, mRNA, miRNA, and many other
    non-coding RNAs. Because \acp{EV} contain mRNA from the cell of
    origin and can be obtained without invasive procedures, they are a
    promising remote sensor of the transcriptome of tissues which
    would otherwise be inaccessible.
    
    One requirement for remotely sensing the transcriptome is
    identifying the source tissue of a specific \ac{EV}. To identify
    the source tissue of a specific \ac{EV}, we have identified
    tissue-specific miRNA and mRNA markers for multiple organs and
    tissues from existing publicly-available RNAseq transcriptomes by
    calculating a tissue specificity index.
    
    The tissue specificity index for gene $g$ ($\tau_g$) is
    
    \begin{equation}
      \tau_g = \frac{\sum_{i=1}^{i=N} \left(1-\frac{X_{i,g}}{\mathrm{max}(X_g)}\right)}{N-1}
    \end{equation}
    \label{eq:tissue_specificity_index}
    
    where $N$ is the number of tissues, $X_{i,g}$ is the mean
    expression of gene $g$ in tissue $i$ and $\mathrm{max}(X_g)$ is
    the maximal expression of gene $g$ across all tissues.
    
    Using machine learning techniques we trained a multiclass \ac{SVM} to classify the tissue of origin.
    
%     Our approach identified MIR1914, MIR126, SNORD12, and SNORD5 as
%     uterus specific. CGB, LGALS13, and members of the PSG family were
%     identified as placenta specific. The accuracy of the classifier
%     was verified using RNAseq transcriptome data not in the training
%     set. We show the applicability of the classifier to simulated EVs
%     by down sampling reads to $10^5$ reads and biasing the reads to
%     account for preferential association of RNA motifs with EV
%     packaging machinery. When coupled with single-EV sequencing, our
%     classifier enables clinicians to monitor disease-specific markers
%     over the course of treatment without biopsies, increasing the
%     ability of clinicians to adapt treatment modalities to the
%     individual patient. 
  }
  
  \block{Methods}{ %
    \Sexpr{star.logs[grepl("Number of input reads",field),sum(as.numeric(value))]} %
    %
    RNAseq reads from %
    % 
    \Sexpr{length(star.logs[grepl("Number of input reads",field),value])} %
    % 
    samples corresponding to %
    \Sexpr{length(unique(na.omit(tissue.specific.genes[,tissue])))} %
    different tissues obtained from the Roadmap Epigenomics
    \cite{Bernstein.ea2010:NIHRoadmapEpigenomicsMapping} and GTEx
    \cite{Gtexconsortium.2015:Thegenotypetissueexpression} projects,
    as well as uterine (GSE50599,
    \cite{Chan.ea2014:Assessmentofmyometrialtranscriptome}) and
    placenta (GSE66622,
    \cite{Hughes.ea2015:Evaluatingintra-andinter-individual}) samples
    from the SRA.
    
    Reads were aligned to \textit{Homo sapiens} reference genome
    GRCh38 with Ensembl annotation release 80 using STAR v2.4.2a
    \cite{Dobin.ea2013:STARultrafastuniversalRNA-seq} and called using
    Tophat 2 v2.0.10
    \cite{Trapnell.ea2009:TopHatdiscoveringsplicejunctions}. The
    average percentage of uniquely mapped reads was %
    \Sexpr{mean(as.numeric(gsub("%","",star.logs[field=="Uniquely mapped reads %",value])))}
    and the average total mapping percentage was %
    \Sexpr{mean(as.numeric(gsub("%","",star.logs[field=="Uniquely mapped reads %",value]))+as.numeric(gsub("%","",star.logs[field=="% of reads mapped to multiple loci",value])))}.
    % 
    For samples where the reads were not publicly available (GTEx and
    some Roadmap Epigenomics samples), the RPKM were taken from
    publicly available result files (see code for details).

    The \ac{SVM} was generated using caret
    \cite{Kuhn2008:Buildingpredictivemodelsinr} and svmRadial
    \cite{Karatzoglou2004:kernlab} with 10 cycles of 10 repeats and a
    tune length of 8 with a training set representing $\approx 75\%$ of the
    samples ($\approx 2450$ samples total, including $\approx 35$
    uterine or $\approx 60$ placenta) in each of the groups chosen at
    random, with the remaining samples making up a testing set
    ($\approx 820$ samples). For comparison purposes, we also
    generated a \ac{KNN} based predictor using the same training and
    testing sets of genes.
    
}


\column{0.25}
  \block{Uterus Specific Genes}{

{\centering
% <<uterus.specific.markers.table,results="asis",echo=FALSE>>=
% temp <- tissue.specific.genes[tissue=="uterus",list(gene,specificity,gi.max)]
% setorder(temp,-gi.max)
% setnames(temp,c("Gene","Tissue Specificity","Mean Expression"))
% print(xtable(temp,
%              caption="Uterus specific genes",label="tab:uterus_specific_genes"),
%       floating=FALSE,
%       include.rownames=FALSE,
%       )
% rm(temp)
% @ 
% }

<<uterus.specific.markers.figure,fig.width=8,fig.height=8,echo=FALSE>>=
print(specific.markers.plot("uterus")
      + xlab("Uterus Specific Genes")
      + ylab("Tissue"))
@
}
\captionof{figure}{Uterus specific genes. $\log_2 \mathrm{FPKM}$ of
  genes with tissue specific indices $\ge 0.98$ which are specific to
  uterus tissue in uterus and tissues with second highest expression
  for those genes. With the exception of FTX\_1, which is expressed in
  multiple fetal tissues and MYH11, a major component of smooth
  muscle, all genes are highly specific to uterus.}

}

  \block{Placenta Specific Genes}{
{\centering
% <<placenta.specific.markers.table,results="asis",echo=FALSE>>=
% temp <- tissue.specific.genes[tissue=="placenta" & gi.max >= 1000,
%                               list(gene,specificity,gi.max)]
% setorder(temp,-gi.max)
% setnames(temp,c("Gene","Tissue Specificity","Mean Expression"))
% print(xtable(temp,
%              caption="Placenta specific genes",label="tab:placenta_specific_genes"),
%       floating=FALSE,
%       include.rownames=FALSE,
%       )
% rm(temp)
% @ 
  
<<placenta.specific.markers.figure,fig.width=8,fig.height=8,echo=FALSE>>=
print(specific.markers.plot("placenta",
                            genes=as.character(tissue.specific.genes[tissue=="placenta" & 
                                                                         gi.max >= 1024
                               ,gene]))
      + xlab("Uterus Specific Genes")
      + ylab("Tissue"))
@

}
\captionof{figure}{Placenta specific genes. $\log_2 \mathrm{FPKM}$ of
  genes with tissue specific indices $\ge 0.98$ and
  $\mathrm{FPKM} \ge 1024$ which are specific to uterus tissue in
  placenta and tissues with second highest expression for those
  genes.}
}

\column{0.25}

\block{SVM to distinguish tissues}{
  % Describe SVM approach
    
  % Figure showing the samples
  
  % Ability to identify placenta

  % Ability to identify uterus

<<placenta.roc.curve,fig.width=8,fig.height=8,echo=FALSE>>=
pl.roc.obj1 <- 
    plot.roc(as.numeric(predictions.caret.svm[["placenta"]][["testing"]][,"actual"])-1,
             as.numeric(predictions.caret.svm[["placenta"]][["testing"]][,"placenta"]),
             percent=TRUE,
             ## main="Training vs Testing set for Uterus",
             col="#002058"
             )
pl.roc.obj2 <- 
    lines.roc(as.numeric(predictions.caret.svm[["placenta"]][["training"]][,"actual"])-1,
              as.numeric(predictions.caret.svm[["placenta"]][["training"]][,"placenta"]),
              percent=TRUE,
              col="#E87722")
# testobj <- roc.test(rocobj1, rocobj2)
# text(50, 50, labels=paste("p-value =", format.pval(testobj$p.value)), adj=c(0, .5))

legend("bottomright",
       legend=c("Placenta Testing Set", "Placenta Training Set"),
       col=c("#002058", "#E87722"),
       lwd=2)
@ 
\captionof{figure}{\ac{ROC} curve of the placenta \ac{SVM}. The
  placenta \ac{SVM} is fairly effective at distinguishing between
  samples of placental origin and non-placental origin.}

  
  % ROC curve for uterus/not uterus

<<uterus.roc.curve,fig.width=8,fig.height=8,echo=FALSE>>=
ut.roc.obj1 <- 
    plot.roc(as.numeric(predictions.caret.svm[["uterus"]][["testing"]][,"actual"])-1,
             as.numeric(predictions.caret.svm[["uterus"]][["testing"]][,"uterus"]),
             percent=TRUE,
             ## main="Training vs Testing set for Uterus",
             col="#002058"
             )
ut.roc.obj2 <- 
    lines.roc(as.numeric(predictions.caret.svm[["uterus"]][["training"]][,"actual"])-1,
              as.numeric(predictions.caret.svm[["uterus"]][["training"]][,"uterus"]),
              percent=TRUE,
              col="#E87722")
# testobj <- roc.test(rocobj1, rocobj2)
# text(50, 50, labels=paste("p-value =", format.pval(testobj$p.value)), adj=c(0, .5))

legend("bottomright",
       legend=c("Uterus Testing Set", "Uterus Training Set"),
       col=c("#002058", "#E87722"),
       lwd=2)
@ 
\captionof{figure}{\ac{ROC} curve of the uterus \ac{SVM}. The uterus \ac{SVM} is
  highly effective at distinguishing between samples of uterus origin and
  non-uterus origin}

}


\column{0.25}

\block{Identification of tissue in reduced reads}{
  % Describe reduced read approach
  
  % 
}

\block{Conclusions}{
  \begin{itemize}
  \item Pregnancy-specific glycoproteins (PSG1-8) are specific to
    placental tissues,and PAEP Progestagen-Associated Endometrial
    Protein is specific to uterus, as previously described.
  \item Numerous Long-noncoding RNA and smaller RNAs are specific to uterus and placenta
  \item \ac{SVM} are able to accurately identify the tissue of origin
    for uterus and placenta.
%   \item Non-maternal variants may also be able to distinguish between
%     placental reads and maternal reads in addition to the approach
%     used here.
  \end{itemize}
}

\block{References}{ 
  
  \begin{center}
    \mbox{}\vspace{-\baselineskip}
  \printbibliography[heading=none]
\end{center}
}

\block{Code and Datasets}{
The code and datasets for this poster can be found at
\url{http://github.com/uiuc-cgm/exosome_markers_per_med_sep_2015.git}
  
% 
}
\block{}{
  \includegraphics[width=\linewidth,height=6cm,keepaspectratio]{figures/uofi_mark}
}

\end{columns}

\end{document}
