#+OPTIONS: ^:{}
#+OPTIONS: author:nil
#+ODT_STYLES_FILE: "isrs_default.odt"

# http://vetmed.illinois.edu/isrs/abstract-guidelines/
#
# ALL Abstracts MUST be submitted as a MS-Word file in the following format:
# 
# Set top, bottom, left and right margins at 1 inch.
# Choose font Arial, size 11, regular.
# All text must be single spaced.
# Type the abstract title in UPPERCASE and make it bold font.
# In the same paragraph, add author names and affiliations after the title, and underline the presenting author.
# Leave one line between the title/author/affiliation paragraph and the abstract body.
# The abstract should be single spaced and justified.
# At the end of the abstract, enter financial support information.
#
# The abstract should not exceed 3000 characters (not including spaces).

*IDENTIFYING EXTRACELLULAR VESICLES ORIGINATING FROM THE PLACENTA AND*
*UTERUS USING RNA EXPRESSION SIGNATURES.* _Don L. Armstrong^{1}_, Monica
Uddin^{1,2}, Derek Wildman.^{1}  1. /Institute for Genomic Biology/,
/Computing Genomes for Reproductive Health, University of Illinois/,
/Urbana-Champaign. 2. Department of Psychology, University of Illinois,
Urbana-Champaign./


Extracellular vesicles (EVs) are small (40-1000nm) compartments that
can be found in blood, saliva, and urine. EVs originate from the
plasma membrane or multivesicular endosome of many cell types and
contain proteins, lipids, mRNA, miRNA, and many other non-coding RNAs.
Because EVs contain mRNA from the cell of origin and can be obtained
without invasive procedures, they are a promising remote sensor of the
transcriptome of tissues which would otherwise be inaccessible. One
requirement for remotely sensing the transcriptome is identifying the
source tissue of a specific EV. To identify the source tissue of a
specific EV, we have identified tissue-characterizing miRNA and mRNA
markers for multiple organs and tissues from existing
publicly-available RNAseq transcriptomes by calculating a tissue
specificity index. Our approach identified FTX_1, PAEP, ZNFX1-AS1_3,
and PVT1_3 as characteristic of uterus and CGA, members of the CSH
family, and members of the PSG family as characteristic of placenta.
Using machine learning techniques we also trained a Support Vector
Machine to classify the tissue of origin. The accuracy of the
classifier was verified using RNAseq transcriptome data not in the
training set. We show the applicability of the classifier to simulated
EVs by down sampling reads and accurately identify placenta with only
10 reads, whereas uterus requires 5Ã—10^{3} reads. When coupled with
single-EV sequencing (or variant calling in whole blood samples for
placental tissue of fetal origin), our classifier enables clinicians
to monitor disease-specific markers over the course of treatment
without biopsies, increasing the ability of clinicians to adapt
treatment modalities to the individual patient.
